downloader/src/main/kotlin/vcs/Git.kt

```diff
@@ -171,27 +123,34 @@ class Git : GitBase() {
         }
 
         // Fall back to fetching everything.
-        log.info { "Trying to fetch everything including tags." }
+        return try {
+            log.info { "Trying to fetch everything including tags." }
 
-        if (workingTree.isShallow()) {
-            run(targetDir, "fetch", "--unshallow", "--tags", "origin")
-        } else {
-            run(targetDir, "fetch", "--tags", "origin")
-        }
+            if (workingTree.isShallow()) {
+                run(workingTree.workingDir, "fetch", "--unshallow", "--tags", "origin")
+            } else {
+                run(workingTree.workingDir, "fetch", "--tags", "origin")
+            }
 
-        revisionCandidates.find { candidate ->
-            try {
-                run(targetDir, "checkout", candidate)
-                true
-            } catch (e: IOException) {
-                e.showStackTrace()
+            run(workingTree.workingDir, "checkout", revision)
+            true
+        } catch (e: IOException) {
+            e.showStackTrace()
 
-                log.info { "Failed to checkout revision '$candidate'. Trying next candidate, if any." }
+            false
```

Comment:
We usually also log a warning in case stacktraces are disabled.

Reply:
Done.